src/components/attendee/RetunableMediaElement/hooks.tsx

```diff
@@ -0,0 +1,51 @@
+import { useMemo } from "react";
+
+import { COLLECTION_RETUNABLE_MEDIA_ELEMENTS } from "settings";
+
+import { useRefiDocument } from "hooks/fire/useRefiDocument";
+
+import {
+  NotTunedSettings,
+  RetunableMediaElementSettings,
+  RetunableMediaSource,
+} from "./types";
+
+interface useRetunableMediaElementDataOptions {
+  spaceId: string;
+}
+
+const NOT_TUNED_SETTINGS: NotTunedSettings = Object.freeze({
+  sourceType: RetunableMediaSource.notTuned,
+});
+
+export const useRetunableMediaElement = ({
+  spaceId,
+}: useRetunableMediaElementDataOptions) => {
+  const queryPath = useMemo(
+    () => [COLLECTION_RETUNABLE_MEDIA_ELEMENTS, spaceId],
+    [spaceId]
+  );
+
+  const {
+    data: settings,
+    isLoading,
+  } = useRefiDocument<RetunableMediaElementSettings>(queryPath);
+
+  if (isLoading) {
+    return {
+      isLoading,
+      settings: NOT_TUNED_SETTINGS,
```

Comment:
I think just freezing this object will not prevent this hook triggering re-renders in cases it's output isn't deconstructed.

I think it's one of those "better be safe than sorry" to add `useMemo()` and do these `if` checks inside it.

Reply:
Resolved in 40c9750b0